cmake/utils.cmake

```diff
@@ -145,3 +145,18 @@ function(list_add_prefix
     endforeach()
     set(${list_out} ${result} PARENT_SCOPE)
 endfunction()
+
+# string() has JOIN option but only since 3.12.
+function(string_join
+    glue
+    string_out
+)
+    set(result "")
+    foreach(item IN LISTS ARGN)
+        set(result "${result}${item}${glue}")
+    endforeach()
+    string(LENGTH ${result} len)
```

Comment:
`${result}` must be enclosed in quotes otherwise the function wouldn't work if the item list is empty.

Anyway, adding the terminating comma just to strip it on the next line looks ugly. Let's not add it in the first place:

```cmake
function(string_join
    glue
    string_out
)
    set(result "")
    set(is_first_item TRUE)
    foreach(item IN LISTS ARGN)
        if(is_first_item)
            set(result "${item}")
            set(is_first_item FALSE)
        else()
            set(result "${result}${glue}${item}")
        endif()
    endforeach()
    set(${string_out} ${result} PARENT_SCOPE)
endfunction()
```

Reply:
This doesn't look perfect either, but I guess it's better, applied your suggestion.